pkg/console/subresource/route/route.go

```diff
@@ -88,6 +101,41 @@ func Validate(route *routev1.Route) (*routev1.Route, bool) {
 	return route, changed
 }
 
+func IsCustomRouteSet(operatorConfig *operatorv1.Console) bool {
+	if len(operatorConfig.Spec.Route.Hostname) != 0 {
+		return true
+	}
+	return false
+}
+
+// Check if reference for secret holding custom serving-cert is set
+func IsCustomRouteSecretSet(operatorConfig *operatorv1.Console) bool {
+	if len(operatorConfig.Spec.Route.Secret.Name) != 0 {
+		return true
+	}
+	return false
+}
+
+func ValidateCustomRoute(operatorConfig *operatorv1.Console, client routeclient.RoutesGetter) error {
+	defaultRoute, err := client.Routes(api.OpenShiftConsoleNamespace).Get(api.OpenShiftConsoleRouteName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	if is := isClusterDomain(operatorConfig.Spec.Route.Hostname, defaultRoute.Spec.Host); !is {
+		if IsCustomRouteSecretSet(operatorConfig) {
+			return fmt.Errorf("secret reference for custom serving-cert is not defined")
+		}
+	}
+
+	return err
+}
+
+func isClusterDomain(customRouteHost, defaultRouteHost string) bool {
```

Comment:
So this is answering "Is the custom host still on the default ingress domain"?

Reply:
yes